xpk.py

```diff
@@ -4889,15 +4889,16 @@ def get_main_container(args, system, docker_image, resource_type) -> str:
 
   tpu_stacktrace_terminate_command = ''
   if (
-      system.accelerator_type == AcceleratorType['TPU']
+      not args.use_pathways
+      and system.accelerator_type == AcceleratorType['TPU']
       and args.deploy_stacktrace_sidecar
   ):
     tpu_stacktrace_terminate_command = (
         'touch /shared-volume/stacktrace_signal; '
     )
 
   xpk_return_user_exit_code = ''
-  if args.restart_on_user_code_failure:
+  if not args.use_pathways and args.restart_on_user_code_failure:
```

Comment:
Do we need many of these `if not args.use_pathways`? It the argument doesn't exist in the create_pathways subfunction, then that prevents this code path right? 

I feel like having several if not args.use_pathways will be hard to maintain.

Reply:
It is necessary to have the condition to not go through this code path. Without the argument in the create_pathways subfunction, the code path is not prevented, but arg not found error is observed.

Reply:
It sounds cleaner to use the arguments set in subfunctions to determine which code paths can be run instead of having to do if checks in the code? What do you think, and let me know if there is some error that prevents this from working!

Reply:
I agree, I tried to keep it as simple and clean as I could. However, some args are not shared (present in create and not in create_pathways). When a user runs create_pathways and XPK goes through these code paths, the args are "not defined" and hence not found. 
Example -
"AttributeError: 'Namespace' object has no attribute 'deploy_stacktrace_sidecar'"

I have hence meticulously abstracted the code paths that are not needed for Pathways.